packages/test-utils/src/utils/mock-client-rects.js

```diff
@@ -0,0 +1,20 @@
+function isHidden( element ) {
+	if ( element.parentElement && isHidden( element.parentElement ) ) {
+		return true;
+	}
+	return (
+		! element.style ||
+		element.style.display === 'none' ||
+		element.style.visibility === 'hidden' ||
+		element.hidden
+	);
+}
+
+window.Element.prototype.getClientRects = function() {
```

Comment:
Should we be modifying the global this way here? Is there even any guarantee that the DOM globals exist? In our own project, we have these globals set up, but this is managed by [the Jest preset](https://github.com/WordPress/gutenberg/blob/master/packages/jest-preset-default/scripts/setup-globals.js). Are other projects which consume from `test-utils` expected to set this up themselves? Should this be documented?

Reply:
Good question! This library will require projects to setup an environment that at least simulates the browser, like `jsdom`, which seems pretty common on React projects (I'm not sure about custom blocks projects though). But I'll include that in the readme.

This mock is needed here because `jsdom` can't do layout and therefore there will be no value for `getClientRects`, `clientWidth` or `clientHeight`. The `isVisible` function in `@wordpress/dom` would always return `false`.

I'm going to experiment mocking that only when the related focus methods are called and restoring it right after that. It seems to be a safer approach.